src/gt4py/next/program_processors/runners/gtfn.py

```diff
@@ -126,7 +139,7 @@ def compilation_hash(otf_closure: stages.ProgramCall) -> int:
     translation=GTFN_GPU_TRANSLATION_STEP,
     bindings=nanobind.bind_source,
     compilation=GTFN_DEFAULT_COMPILE_STEP,
-    decoration=convert_args,
+    decoration=functools.partial(convert_args, for_cp=True),
```

Comment:
use device instead of for_cp

Reply:
You mean rename the argument?

Reply:
Or also change the type to `DeviceType` and change the logic accordingly?

Reply:
sorry, that note was for myself. Enrique, suggested instead of using the boolean, and do the same logic if DeviceType==GPU

Reply:
I can quickly do that.